src/compile-uri/compile-params.coffee

```diff
@@ -1,33 +1,30 @@
 fury = require('fury')
+{Element} = require('minim')
+JSON06Serialiser = require('minim/lib/serialisers/json-0.6')
 
 {content} = require('../refract')
 
 
+# Convert a Href Variables element to a Dredd Representation
+# Accepts both hrefVariables minim element, and 0.6 serialised Refract hrefVariables
 module.exports = (hrefVariables) ->
   parameters = {}
 
-  for member in content(hrefVariables) or []
-    {key, value} = content(member)
-
-    name = content(key)
-    types = (content(member.attributes?.typeAttributes) or [])
-
-    if value?.element is 'enum'
-      if value.attributes?.samples?.length and value.attributes?.samples[0].length
-        exampleValue = content(value.attributes.samples[0][0])
-      else
-        exampleValue = content(content(value)[0])
-      if value.attributes?.default?.length
-        defaultValue = content(value.attributes.default[0])
-    else
-      exampleValue = content(value)
-      if value.attributes?.default
-        defaultValue = content(value.attributes?.default)
-
-    parameters[name] =
-      required: 'required' in types
-      default: defaultValue
-      example: exampleValue
-      values: if value?.element is 'enum' then ({value: content(v)} for v in content(value)) else []
+  if hrefVariables
+    if not (hrefVariables instanceof Element)
+      serialiser = new JSON06Serialiser(fury.minim)
+      hrefVariables = serialiser.deserialise(hrefVariables)
+
+    hrefVariables.forEach((value, key, member) ->
+      name = key.toValue()
+      typeAttributes = member.attributes.get('typeAttributes')?.toValue() or []
+      values = value.attributes.get('enumerations')?.toValue() or []
+
+      parameters[name] =
+        required: 'required' in typeAttributes
+        default: value.attributes.get('default')?.toValue()
+        example: value.toValue() or values[0]
```

Comment:
The problem lies around the example and default values. For an array these would've been an array of elements. The function returned example and default as an array of elements and thus expansion would expand these array of element objects instead of array of values.